go/cmd/dolt/commands/batsee.go

```diff
@@ -0,0 +1,377 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package commands
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"sync"
+	"syscall"
+	"time"
+
+	"github.com/fatih/color"
+
+	"github.com/dolthub/dolt/go/cmd/dolt/cli"
+	"github.com/dolthub/dolt/go/cmd/dolt/errhand"
+	"github.com/dolthub/dolt/go/libraries/doltcore/env"
+	"github.com/dolthub/dolt/go/libraries/utils/argparser"
+)
+
+var batseeDoc = cli.CommandDocumentationContent{
+	ShortDesc: `Run the Bats Tests concurrently`,
+	LongDesc:  `From within the integration-test/bats directory, run the bats tests concurrently. Output for each test is written to a file in the batsee_output directory.`,
+	Synopsis: []string{
+		`-t 42`,
+		`--skip-slow --max-time 1h15m`,
+		`--retries 2 --only types.bats,foreign-keys.bats`,
+	},
+}
+
+type BatseeCmd struct {
+	messages []string
+}
+
+var _ cli.Command = &BatseeCmd{}
+var _ cli.RepoNotRequiredCommand = &BatseeCmd{}
+
+func (b BatseeCmd) Name() string {
+	return "batsee"
+}
+
+func (b BatseeCmd) Description() string {
+	return batseeDoc.ShortDesc
+}
+func (b BatseeCmd) Docs() *cli.CommandDocumentation {
+	return cli.NewCommandDocumentation(batseeDoc, b.ArgParser())
+}
+
+func (b BatseeCmd) Hidden() bool {
+	return true
+}
+
+func (b BatseeCmd) ArgParser() *argparser.ArgParser {
+	ap := argparser.NewArgParserWithMaxArgs(b.Name(), 0)
+	ap.SupportsUint("threads", "t", "threads", "Number of tests to execute in parallel. Defaults to 12")
+	ap.SupportsFlag("skip-slow", "s", "Skip slow tests. This is a static list of test we know are slow, may grow stale.")
+	ap.SupportsString("max-time", "", "", "Maximum time to run tests. Defaults to 30m")
+	ap.SupportsString("only", "", "", "Only run the specified test, or tests (comma separated)")
+	ap.SupportsInt("retries", "r", "retries", "Number of times to retry a failed test. Defaults to 1")
+	return ap
+}
+
+func (b BatseeCmd) RequiresRepo() bool {
+	return false
+}
+
+type batsResult struct {
+	runtime time.Duration
+	path    string
+	err     error
+	skipped bool
+	aborted bool
+}
+
+// list of slow commands. These tend to run more than 5-7 min, so we want to run them first.
+var slowCommands = map[string]bool{
+	"types.bats":                 true,
+	"keyless.bats":               true,
+	"index-on-writes.bats":       true,
+	"constraint-violations.bats": true,
+	"foreign-keys.bats":          true,
+	"index.bats":                 true,
+	"sql-server.bats":            true,
+	"index-on-writes-2.bats":     true,
+	"sql.bats":                   true,
+	"remotes.bats":               true,
+}
+
+func (b BatseeCmd) Exec(ctx context.Context, commandStr string, args []string, dEnv *env.DoltEnv, cliCtx cli.CliContext) int {
+	ap := b.ArgParser()
+	help, usage := cli.HelpAndUsagePrinters(cli.CommandDocsForCommandString(commandStr, batseeDoc, ap))
+
+	apr, err := ap.Parse(args)
```

Comment:
You can simplify this with

```suggestion
	ap := cmd.ArgParser()
	help, usage := cli.HelpAndUsagePrinters(cli.CommandDocsForCommandString(commandStr, batseeDoc, ap))
	apr := cli.ParseArgsOrDie(ap, args, help)
```

Reply:
There is a more general problem which is the differentiation between `help` and `usage` - I'm not convinced this is required, but the error handling here was trying to preserve a difference. I'll do it the way you suggest, but there is room for general clean up across the board here.